sfff/src/font.rs

```diff
@@ -0,0 +1,417 @@
+    /// tuple brackets)
+    pub hairpin_thickness: i32,
+    ///
+    pub rehearsal_box_thickness: i32,
+
+    // Glyph metadata (Notehead & Stem Positions)
+    pub notehead: [[i32; 2]; 2], // also includes slashed notehead
+    pub notehead_x: [[i32; 2]; 2],
+    pub notehead_diamond: [[i32; 2]; 2],
+    pub notehead_triangle: [[i32; 2]; 2],
+    pub notehead_slash: [[i32; 2]; 2],
+
+    pub notehead_half: [[i32; 2]; 2], // also includes slashed notehead
+    pub notehead_half_x: [[i32; 2]; 2],
+    pub notehead_half_diamond: [[i32; 2]; 2],
+    pub notehead_half_triangle: [[i32; 2]; 2],
+    pub notehead_half_slash: [[i32; 2]; 2],
+
+    pub notehead_whole: [[i32; 2]; 2], // also includes slashed notehead
+    pub notehead_whole_x: [[i32; 2]; 2],
+    pub notehead_whole_diamond: [[i32; 2]; 2],
+    pub notehead_whole_triangle: [[i32; 2]; 2],
+    pub notehead_whole_slash: [[i32; 2]; 2],
+
+    pub notehead_double: [[i32; 2]; 2], // also includes slashed notehead
+    pub notehead_double_x: [[i32; 2]; 2],
+    pub notehead_double_diamond: [[i32; 2]; 2],
+    pub notehead_double_triangle: [[i32; 2]; 2],
+    pub notehead_double_slash: [[i32; 2]; 2],
+}
+
+/// Read one 16-bit value in LE byte order
+fn read_u16<T: Read>(reader: &mut T) -> Result<u16, ReadError> {
+    let mut buf = [0u8; 2];
+    reader
+        .read_exact(&mut buf)
+        .map_err(|_| ReadError::UnexpectedEOF)?;
+    Ok(u16::from_le_bytes(buf))
+}
+
+/// Write one 16-bit value in LE byte order
+fn write_u16<T: Write>(writer: &mut T, value: u16) -> Result<(), WriteError> {
+    let bytes = writer
+        .write(&value.to_le_bytes())
+        .map_err(|_| WriteError::Prevented)?;
+    // FIXME: turn this into a write error
+    assert_eq!(bytes, 2);
+    Ok(())
+}
+
+/// Read one 32-bit value in LE byte order
+fn read_i32<T: Read>(reader: &mut T) -> Result<i32, ReadError> {
+    let mut buf = [0u8; 4];
+    reader
+        .read_exact(&mut buf)
+        .map_err(|_| ReadError::UnexpectedEOF)?;
+    Ok(i32::from_le_bytes(buf))
+}
+
+/// Write one 32-bit value in LE byte order
+fn write_i32<T: Write>(writer: &mut T, value: i32) -> Result<(), WriteError> {
+    let bytes = writer
+        .write(&value.to_le_bytes())
+        .map_err(|_| WriteError::Prevented)?;
+    // FIXME: turn this into a write error
+    assert_eq!(bytes, 4);
+    Ok(())
+}
+
+/// Read two X,Y positions in LE byte order
+fn read_positions<T: Read>(reader: &mut T) -> Result<[[i32; 2]; 2], ReadError> {
+    let x1 = read_i32(reader)?;
+    let y1 = read_i32(reader)?;
+    let x2 = read_i32(reader)?;
+    let y2 = read_i32(reader)?;
+    Ok([[x1, y1], [x2, y2]])
+}
+
+/// Write two X,Y positions in LE byte order
+fn write_positions<T: Write>(
+    writer: &mut T,
+    pos: [[i32; 2]; 2],
+) -> Result<(), WriteError> {
+    write_i32(writer, pos[0][0])?;
+    write_i32(writer, pos[0][1])?;
+    write_i32(writer, pos[1][0])?;
+    write_i32(writer, pos[0][1])?;
```

Comment:
Maybe make it match the read method with destructuring?

```suggestion
    let [[x1, y1], [x2, y2]] = pos;
    write_i32(writer, x1)?;
    write_i32(writer, y1)?;
    write_i32(writer, x2)?;
    write_i32(writer, y2)?;
```